src/parser/FunctionParser.mts

```diff
@@ -1,14 +1,19 @@
 import { IsSimpleParameterList } from '../static-semantics/all.mjs';
-import { getDeclarations } from './Scope.mjs';
+import { ArrowInfoStack, getDeclarations } from './Scope.mjs';
 import { Token } from './tokens.mjs';
 import { IdentifierParser } from './IdentifierParser.mjs';
+import type { ParseNode } from './Parser.mjs';
 
-export const FunctionKind = {
-  NORMAL: 0,
-  ASYNC: 1,
-};
+export enum FunctionKind {
```

Comment:
what does this compile to? will it be missing in the public api now?

Reply:
This is compiled using babel's TS support.

```ts
enum FunctionKind {
   Sync, Async,
}
```

goes to

```js
"use strict";

var FunctionKind;
(function (FunctionKind) {
  FunctionKind[FunctionKind["Sync"] = 0] = "Sync";
  FunctionKind[FunctionKind["Async"] = 1] = "Async";
})(FunctionKind || (FunctionKind = {}));
```